azure-configurationservice/azure/configuration/aio/azure_configuration_client_async.py

```diff
@@ -0,0 +1,262 @@
+            key, etag, **kwargs
+        )
+        current_configuration_setting = await self.get_configuration_setting(key, label)
+        if value is not None:
+            current_configuration_setting.value = value
+        if content_type is not None:
+            current_configuration_setting.content_type = content_type
+        if tags is not None:
+            current_configuration_setting.tags = tags
+        return await self._impl.create_or_update_configuration_setting(
+            configuration_setting=current_configuration_setting,
+            key=key,
+            label=label,
+            headers=custom_headers,
+            error_map={404: ResourceNotFoundError, 412: ResourceModifiedError},
+        )
+
+    async def set_configuration_setting(
+        self, configuration_setting, **kwargs
+    ):  # type: (ConfigurationSetting, dict) -> ConfigurationSetting
+        """
+        Example
+
+        .. code-block:: python
+
+            # in async function
+            config_setting = ConfigurationSetting(
+                key="MyKey",
+                label="MyLabel",
+                value="my set value",
+                content_type="my set content type",
+                tags={"my set tag": "my set tag value"}
+            )
+            returned_config_setting = await async_client.set_configuration_setting(config_setting)
+        """
+        return await super().set_configuration_setting(configuration_setting, **kwargs)
+
+    async def delete_configuration_setting(
+        self, key, label=None, etag=None, **kwargs
+    ):  # type: (str, str, str, dict) -> ConfigurationSetting
+        """
+        Example
+
+        .. code-block:: python
+
+            # in async function
+            deleted_config_setting = await async_client.delete_configuration_setting(
+                key="MyKey", label="MyLabel"
+            )
+        """
+        return await super().delete_configuration_setting(key, label, etag, **kwargs)
+
+    def list_revisions(
+        self, labels=None, keys=None, accept_date_time=None, fields=None, **kwargs
+    ):  # type: (list, list, datetime, list, dict) -> Paged
+        """
+        Example
+
+        .. code-block:: python
+
+            # in async function
+            from datetime import datetime, timedelta
+
+            accept_date_time = datetime.today() + timedelta(days=-1)
+
+            all_revisions = async_client.list_revisions()
+            async for item in all_revisions:
+                pass  # do something
+
+            filtered_revisions = async_client.list_revisions(
+                labels=["*Labe*"], keys=["*Ke*"], accept_date_time=accept_date_time
+            )
+            async for item in filtered_revisions:
+                pass  # do something
+        """
+        return super().list_revisions(labels, keys, accept_date_time, fields, **kwargs)
+
+
+AzureConfigurationClient.__doc__ = (
```

Comment:
We need to revisit this. While this works for intellisense and doc generation, it doesn't make browsing the source code a very pleasant experience... I'm fine with this for now, but we should push this onto the backlog...

Reply:
Yes, I'm also thinking whether this is the a good way. Duplicate could be the best way